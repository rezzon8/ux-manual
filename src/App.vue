<template>
  <v-app>
    <v-overlay :value="$store.state.is_data_processing">
      <v-progress-circular indeterminate size="64"></v-progress-circular>
    </v-overlay>

    <v-app-bar app color="primary" class="elevation-8">
      <v-icon size="26">mdi-book-open-page-variant</v-icon>
      <v-toolbar-title class="ml-3">
        <h2>The UX Manual</h2>
      </v-toolbar-title>

      <v-spacer></v-spacer>

      <template v-if="!userIsAuthenticated">
        <v-btn text @click.stop="loginDialog = true">
          Login
        </v-btn>

        <v-btn text @click.stop="signUpDialog = true">
          Signup
        </v-btn>
      </template>
      <template v-else>
        <v-btn text @click.stop="logOut">
          Logout
        </v-btn>
      </template>
    </v-app-bar>

    <v-main>
      <router-view></router-view>
    </v-main>

    <v-dialog v-model="loginDialog" width="800">
      <v-card>
        <v-card-title class="headline" style="position:relative;">
          Login
        </v-card-title>
        <v-card-text class="pt-5">
          <LoginForm @login="logIn" />
        </v-card-text>
        <v-divider></v-divider>
        <v-card-actions>
          <v-spacer></v-spacer>
          <v-btn @click.stop="loginDialog = false">
            Close
          </v-btn>
        </v-card-actions>
      </v-card>
    </v-dialog>

    <v-dialog v-model="signUpDialog" width="800">
      <v-card>
        <v-card-title class="headline" style="position:relative;">
          Signup
        </v-card-title>
        <v-card-text class="pt-5">
          <SignUpForm @signup="signUp" />
        </v-card-text>
        <v-divider></v-divider>
        <v-card-actions>
          <v-spacer></v-spacer>
          <v-btn @click.stop="signUpDialog = false">
            Close
          </v-btn>
        </v-card-actions>
      </v-card>
    </v-dialog>
  </v-app>
</template>

<script>
import { mapGetters, mapActions } from "vuex";
import LoginForm from "@/components/forms/LoginForm.vue";
import SignUpForm from "@/components/forms/SignUpForm.vue";

export default {
  name: "UX_manual",
  components: {
    SignUpForm,
    LoginForm
  },
  data() {
    return {
      color: "blue",
      loginDialog: false,
      signUpDialog: false
    };
  },
  computed: {
    ...mapGetters(["getDialogState", "userIsAuthenticated"])
  },
  methods: {
    ...mapActions(["LOGIN"]),
    signUp(payload) {
      // signup
      this.$store
        .dispatch("SIGN_UP", payload)
        .then(() => {
          this.signUpDialog = false;
        })
        .catch(e => {
          console.log(e, "error in sign up");
        });
    },
    logIn(payload) {
      // login
      this.$store
        .dispatch("LOGIN", payload)
        .then(() => {
          this.loginDialog = false;
        })
        .catch(e => {
          console.log(e, "error in login");
        });
    },
    logOut() {
      this.$store
        .dispatch("LOGOUT")
        .then(() => {
          // logged out
        })
        .catch(e => {
          console.log(e, "error in logout");
        });
    }
  },
  beforeCreate() {
    this.$store
      .dispatch("GET_DATA")
      .then(() => {
        // data retrieved
      })
      .catch(e => {
        console.log(e.message);
      });
  }
};
</script>

<style lang="scss" scoped>
.v-card__text {
  padding-top: 20px !important;
}
</style>
