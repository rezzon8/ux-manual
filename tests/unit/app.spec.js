import { createLocalVue, mount } from "@vue/test-utils";
import Vue from "vue";
import Vuex from "vuex";
import Vuetify from "vuetify";
import VueRouter from "vue-router";
import Vuelidate from "vuelidate";
import App from "@/App.vue";

Vue.use(Vuelidate);
Vue.use(Vuetify); // added to prevent vuetify lint errors

const localVue = createLocalVue();
const vuetify = new Vuetify();

let wrapper;
let router;
let store;
let state;
let getters;
let actions;
let mutations;

localVue.use(VueRouter);
localVue.use(Vuex);

router = new VueRouter();

const factory = () => {
  return mount(App, {
    propsData: {},
    data: () => ({
      email: "test@test.com",
      password: "12345678",
      confirmPassword: "12345678",
      showPassword: false
    }),
    mocks: {},
    stubs: {},
    store,
    vuetify,
    router,
    localVue
  });
};

describe("App.vue", () => {
  beforeEach(() => {
    state = {
      key: {}
    };
    getters = {
      userIsAuthenticated: jest.fn(false)
    };
    actions = {
      GET_DATA: jest.fn(),
      LOGIN: jest.fn(),
      SIGN_UP: jest.fn(),
      LOGOUT: jest.fn()
    };
    mutations = {
      someMutation: jest.fn()
    };
    store = new Vuex.Store({
      state,
      getters,
      actions,
      mutations
    });
    wrapper = factory();
  });

  afterEach(() => {
    wrapper.destroy();
  });
  it("should render", () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
  it("should define a signUp method", () => {
    expect(wrapper.vm.signUp).toBeDefined();
  });
  it("should define a logIn method", () => {
    expect(wrapper.vm.logIn).toBeDefined();
  });
  it("should define a logOut method", () => {
    expect(wrapper.vm.logOut).toBeDefined();
  });
  it("should dispatch SIGN_UP action", () => {
    const spy = jest.spyOn(actions, "SIGN_UP");
    wrapper.vm.signUp();
    expect(spy).toBeCalled();
  });
  it("should reject SIGN_UP action", () => {
    wrapper.vm.$store._actions.SIGN_UP[0] = jest.fn(() => {
      return Promise.reject();
    });
    wrapper.vm.signUp();
  });
  it("should dispatch LOGIN action", () => {
    const spy = jest.spyOn(actions, "LOGIN");
    wrapper.vm.logIn();
    expect(spy).toBeCalled();
  });
  it("should reject LOGIN action", () => {
    wrapper.vm.$store._actions.LOGIN[0] = jest.fn(() => {
      return Promise.reject();
    });
    wrapper.vm.logIn();
  });
  it("should dispatch LOGOUT action", () => {
    const spy = jest.spyOn(actions, "LOGOUT");
    wrapper.vm.logOut();
    expect(spy).toBeCalled();
  });
  it("should reject LOGOUT action", () => {
    wrapper.vm.$store._actions.LOGOUT[0] = jest.fn(() => {
      return Promise.reject();
    });
    wrapper.vm.logOut();
  });
  it("should reject GET_DATA action", () => {
    actions = {
      ...actions,
      GET_DATA: jest.fn(() => {
        return Promise.reject();
      })
    };
    store = new Vuex.Store({
      getters,
      actions
    });
    wrapper = factory();
    wrapper.vm.logOut();
  });
});
