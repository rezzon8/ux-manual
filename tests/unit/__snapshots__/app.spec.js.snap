// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should render 1`] = `
<div data-app="true" class="v-application v-application--is-ltr theme--light" id="app">
  <div class="v-application--wrap">
    <div class="v-overlay v-overlay--active theme--dark" style="z-index: 5;">
      <div class="v-overlay__scrim" style="opacity: 0.46; background-color: rgb(33, 33, 33); border-color: #212121;"></div>
      <div class="v-overlay__content">
        <div role="progressbar" aria-valuemin="0" aria-valuemax="100" class="v-progress-circular v-progress-circular--indeterminate" style="height: 64px; width: 64px;"><svg xmlns="http://www.w3.org/2000/svg" viewBox="21.333333333333332 21.333333333333332 42.666666666666664 42.666666666666664" style="transform: rotate(0deg);">
            <circle fill="transparent" cx="42.666666666666664" cy="42.666666666666664" r="20" stroke-width="2.6666666666666665" stroke-dasharray="125.664" stroke-dashoffset="125.66370614359172px" class="v-progress-circular__overlay"></circle>
          </svg>
          <div class="v-progress-circular__info"></div>
        </div>
      </div>
    </div>
    <header class="elevation-8 v-sheet theme--light v-toolbar v-app-bar v-app-bar--fixed primary" style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
      <div class="v-toolbar__content" style="height: 64px;"><i aria-hidden="true" class="v-icon notranslate mdi mdi-book-open-page-variant theme--light" style="font-size: 26px;"></i>
        <div class="v-toolbar__title ml-3">
          <h2>The UX Manual</h2>
        </div>
        <div class="spacer"></div> <button type="button" class="v-btn v-btn--flat v-btn--text theme--light v-size--default" id="logIn"><span class="v-btn__content">
        Login
      </span></button> <button type="button" class="v-btn v-btn--flat v-btn--text theme--light v-size--default" id="signUp"><span class="v-btn__content">
        Signup
      </span></button>
      </div>
    </header>
    <main class="v-main" style="padding: 64px 0px 0px 0px;">
      <div class="v-main__wrap"></div>
    </main>
    <div role="dialog" class="v-dialog__container">
      <!---->
    </div>
    <div role="dialog" class="v-dialog__container">
      <!---->
    </div>
  </div>
</div>
`;
